cosmwasm-contracts/marketing/src/contract.rs

```diff
@@ -0,0 +1,273 @@
+    // Replace all live_collections with the new ones
+    self.live_collections.clear(ctx.deps.storage);
+    for (index, live_collection) in live_collections.into_iter().enumerate() {
+        self.live_collections.save(ctx.deps.storage, index as u64, &live_collection)?;
+    }
+
+    Ok(Response::new().add_attributes(attributes))
+  }
+
+  #[msg(exec)]
+  // Only the admin can execute it
+  pub fn update_upcoming_collections(
+    &self,
+    ctx: ExecCtx,
+    upcoming_collections: Vec<MarketingCollectionPreview>,
+  ) -> Result<Response, ContractError> {
+    let attributes = vec![attr("action", "update_upcoming_collections")];
+    let config = self.config.load(ctx.deps.storage)?;
+    // Permission check
+    if ctx.info.sender != config.admin_addr {
+      return Err(ContractError::Unauthorized);
+    }
+    // Replace all upcoming_collections with the new ones
+    self.upcoming_collections.clear(ctx.deps.storage);
+    for (index, upcoming_collection) in upcoming_collections.into_iter().enumerate() {
+        self.upcoming_collections.save(ctx.deps.storage, index as u64, &upcoming_collection)?;
+    }
+
+    Ok(Response::new().add_attributes(attributes))
+  }
+
+  #[msg(exec)]
+  // Only the admin can execute it
+  pub fn update_highlighted_collections(
+    &self,
+    ctx: ExecCtx,
+    highlighted_collections: Vec<MarketingCollectionPreview>,
+  ) -> Result<Response, ContractError> {
+    let attributes = vec![attr("action", "update_highlighted_collections")];
+    let config = self.config.load(ctx.deps.storage)?;
+    // Permission check
+    if ctx.info.sender != config.admin_addr {
+      return Err(ContractError::Unauthorized);
+    }
+    // Replace all highlighted_collections with the new ones
+    self.highlighted_collections.clear(ctx.deps.storage);
+    for (index, highlighted_collection) in highlighted_collections.into_iter().enumerate() {
+        self.highlighted_collections.save(ctx.deps.storage, index as u64, &highlighted_collection)?;
+    }
+
+    Ok(Response::new().add_attributes(attributes))
+  }
+
+  #[msg(query)]
+  pub fn get_config(&self, ctx: QueryCtx) -> StdResult<Config> {
+    let config = self.config.load(ctx.deps.storage)?;
+    Ok(config)
+  }
+
+  #[msg(query)]
+  pub fn get_banners(&self, ctx: QueryCtx) -> StdResult<Vec<Banner>> {
+    let banners: Vec<Banner> = self
+    .banners
+    .range(ctx.deps.storage, None, None, cosmwasm_std::Order::Ascending)
+    .map(|item| item.map(|(_, banner)| banner))
+    .collect::<StdResult<Vec<Banner>>>()?; 
+
+    Ok(banners)
+  }
+
+  #[msg(query)]
+  pub fn get_news(&self, ctx: QueryCtx) -> StdResult<Vec<News>> {
+    let news: Vec<News> = self
+    .news
+    .range(ctx.deps.storage, None, None, cosmwasm_std::Order::Ascending)
+    .map(|item| item.map(|(_, news_item)| news_item))
+    .collect::<StdResult<Vec<News>>>()?; 
+
+    Ok(news)
+  }
+
+  #[msg(query)]
+  pub fn get_live_collections(&self, ctx: QueryCtx) -> StdResult<Vec<MarketingCollectionPreview>> {
```

Comment:
Should we merge 3 queries: live, highlighted, upcoming => 1 query function and use a switch case to return the according data ? we can prevent the repeated code => maintain easier, contract size smaller,  easy to add more query with the same kind of data

Reply:
After reflexion, handling "Which collections are shown in the Launchpad" doesn't make sense.
The collections shown are handled by the Marketplace, automatically, no need to choose.
I'll remove the collections stuff from the marketing contract

Reply:
It's very nice that you are working!